registrations/tests/test_price_group_delete.py

```diff
@@ -0,0 +1,267 @@
+from decimal import Decimal
+from typing import Union
+
+import pytest
+from django.db.models import ProtectedError
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from audit_log.models import AuditLogEntry
+from events.tests.utils import versioned_reverse as reverse
+from helevents.tests.factories import UserFactory
+from registrations.models import PriceGroup
+from registrations.tests.factories import (
+    PriceGroupFactory,
+    RegistrationPriceGroupFactory,
+)
+
+# === util methods ===
+
+
+def delete_price_group(api_client: APIClient, price_group_pk: Union[str, int]):
+    url = reverse(
+        "pricegroup-detail",
+        kwargs={"pk": price_group_pk},
+    )
+
+    return api_client.delete(url)
+
+
+def assert_delete_price_group(api_client: APIClient, price_group_pk: Union[str, int]):
+    response = delete_price_group(api_client, price_group_pk)
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+
+    return response
+
+
+def assert_delete_default_price_group(api_client: APIClient, price_group: PriceGroup):
+    assert PriceGroup.objects.count() == 9
+
+    assert_delete_price_group(api_client, price_group.pk)
+
+    assert PriceGroup.objects.count() == 8
```

Comment:
Hmm, not sure what actually ties this function to default price groups, and thus why this is even needed. Couldn't the count checks be added to `assert_delete_price_group()` and only that function used? 🙂

Reply:
Improved this and similar functions in the other price group test files